evalml/tests/pipeline_tests/test_component_graph.py

```diff
@@ -512,15 +512,15 @@ def test_fit_correct_inputs(
     X = pd.DataFrame(X)
     y = pd.Series(y)
     graph = {"Imputer": [Imputer], "OHE": [OneHotEncoder, "Imputer.x", "Imputer.y"]}
-    expected_x = pd.DataFrame(index=X.index, columns=X.columns).fillna(1)
```

Comment:
The original data has float-type features but the mock returns int-type features. This meant we were technically changing the dtypes between components. The component graph previously changed the int-types back to floats with this code block:

```python
  logical_types = {}
  logical_types.update(X.ww.logical_types)
  for x_input in x_inputs:
      if x_input.ww.schema is not None:
          logical_types.update(x_input.ww.logical_types)
  return_x = _retain_custom_types_and_initalize_woodwork(logical_types, return_x)
```
which is why we need the `astype("float64")` call below.

Since we now use `ww.concat_columns`, we would be preserving the int-types returned by the mocks because that's `ww.concat_columns` is supposed to do! 

So in order to get this test to pass, we either need to make the mocks return floats or get rid of the `astype` call below. I opted for the former because it makes it clearer that the component graph does not overwrite the types returned by the components.